lib/config/browser-options.js

```diff
@@ -30,6 +30,30 @@ exports.getPerBrowser = () => {
                 }
             },
             map: (value, config) => _.extend({}, config.desiredCapabilities, value)
+        }),
+        screenshotMode: option({
+            defaultValue: 'auto',
+            validate: (value) => {
+                is('string', 'screenshotMode')(value);
+
+                if (!_.includes(['fullpage', 'viewport', 'auto'], value)) {
+                    throw new Error('"screenshotMode" must be "fullpage", "viewport" or "auto"');
+                }
+            },
+            map: (value) => {
```

Comment:
This didn't come out as clean as we expected -- I had to pass original config options objects as an argument `originalOptions` above. Couldn't figure out how to access `platformName` or `browserName` otherwise.

Here, in `map(value, config, currentNode)`, expression `config === currentNode` turns out to be `true`. It means that inside the `map` function we only have access to the whole config, but we can't tell which browser we're currently dealing with.